smartsim/launchable/job.py

```diff
@@ -158,3 +160,44 @@ def __str__(self) -> str:  # pragma: no cover
         string = f"SmartSim Entity: {self.entity}\n"
         string += f"Launch Settings: {self.launch_settings}"
         return string
+
+
+@t.final
+class Record:
+    """A Record object to track a launched job along with its assigned
```

Comment:
does it really "track" the job or would something like `A unique identifier for a launched job` express what it does?

Reply:
Good point! I updated the docstring to be a bit more descriptive of the composition of the object!